tooling/storybook-addon/src/components/Tool.tsx

```diff
@@ -0,0 +1,33 @@
+import React, { useCallback } from "react"
+import { useAddonState } from "@storybook/api"
+import { Icons, IconButton } from "@storybook/components"
+import { addons } from "@storybook/addons"
+import { ADDON_ID, TOOL_ID, EVENTS } from "../constants"
+
+export const Tool = () => {
+  const persistedColorMode =
+    localStorage.getItem("chakra-ui-color-mode") === "dark"
```

Comment:
```suggestion
  const isDarkMode =
    localStorage.getItem("chakra-ui-color-mode") === "dark"
```

Reply:
I think its better to move local storage access to useEffect. Otherwise this key will be fetched from localstorage on every render of this component

Reply:
Yes, it could be a useEffect with an empty deps array.